src/components/UserSelector.tsx

```diff
@@ -2,15 +1,77 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classnames from 'classnames';
+import { getUsers } from '../api/users';
+
+type Props = {
+  selectedUser: User | null;
+  onSelectUser: (user: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectUser,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [error, setError] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const loadUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch {
+      setError(true);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
Great job on using `try/catch` for error handling as mentioned in the checklist. This is a good practice to handle potential errors that might occur during the execution of async operations.